Document IsFromUnhandledException/IsFromTerminalException API (.NET) (#18676)
getsentry/sentry-docs: docs/platforms/dotnet/common/configuration/filtering.mdx

Note also that breadcrumbs can be filtered, as discussed in [our Breadcrumbs documentation](/product/issues/issue-details/breadcrumbs/).

#### Filtering by Unhandled or Terminal Exceptions

> When filtering events in <PlatformIdentifier name="before-send" />, you'll often want to treat unhandled exceptions differently from handled ones. For example, you might drop noisy handled exceptions (like network timeouts) while always sending exceptions that went unhandled and crashed your app.

Author: ```suggestion
When filtering events in <PlatformIdentifier name="before-send" />, you may want to treat unhandled exceptions differently from handled ones. For example, you might drop noisy exceptions (like network timeouts) if these were handled, but capture them if they crashed your app.
```

> The SDK provides two extension methods on `SentryEvent` to make this decision:

Author: ```suggestion
The SDK provides two extension methods on `SentryEvent` to enable this:
```

The SDK provides two extension methods on `SentryEvent` to enable this:

- `IsFromUnhandledException()` returns `true` when the event came from an exception that wasn't caught by your application code, but was instead captured through an unhandled exception hook (such as `AppDomain.UnhandledException`, ASP.NET Core middleware, or another integration).
- `IsFromTerminalException()` returns `true` only when the unhandled exception caused the application to crash. Some integrations capture unhandled exceptions that don't terminate the process (for example, unobserved task exceptions) and mark them as non-terminal.

```csharp
options.SetBeforeSend((sentryEvent, hint) =>
{
    // Always send unhandled exceptions, even if they'd be filtered below
    if (sentryEvent.IsFromUnhandledException())
    {
        return sentryEvent;
    }

    // Drop noisy handled exceptions
    if (sentryEvent.Exception is HttpRequestException)
    {
        return null; // Don't send this event to Sentry
    }

    return sentryEvent;
});
```

```fsharp
open Sentry

> open System.Net.Http

Reviewer: TIL: F# has no implicit usings / opens

> options.SetBeforeSend(fun sentryEvent hint ->

Reviewer: question/suggestion: consistency of delegate parameter names with example above

Here, the F# parameters `sentryEvent hint` are consistent with the C# code `(sentryEvent, hint)`.

But in the example above: "Using `BeforeSend`", the C# code also uses `(sentryEvent, hint)` (so we are consistent there) ... but the F# code uses `fun event hint ->` (where we are not consistent).

I believe using "sentryEvent" here is OK, but perhaps we might want to rename the existing F# sample to "sentryEvent" rather than just "event", too. Considering https://learn.microsoft.com/dotnet/fsharp/language-reference/keyword-reference which documents that `event` is "a token that is reserved as keyword for future expansion of F#".

But to be fair ... the inconsistency of `sentryEvent` vs the above `event` is definitely nit-picky ... and not directly related to THIS changeset.
